src/App.tsx

```diff
@@ -1,73 +1,99 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [showDropdown, setShowDropdown] = useState(true);
+  const [title, setTitle] = useState<Person | null>(null);
+
+  const filteredPeople = [...peopleFromServer].filter(person => {
+    const prettyQuery = appliedQuery.toLowerCase().trimStart();
+
+    return person.name.toLowerCase().includes(prettyQuery);
+  });
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
```

Comment:
Your variant is ok, but also it is possible:
```suggestion
  const applyQuery = useCallback(debounce(setAppliedQuery, 300), [setAppliedQuery]);
```